var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1066,10 +1066,14 @@ def libs(self):
         # The +shared variant isn't reliable, as `spack external find` currently can't
         # detect it. If +shared, prefer the shared libraries, but check for static if
         # those aren't found. Vice versa for ~shared.
-        if "+shared" in self.spec:
-            candidates = shared_libs + static_libs
-        else:
+        #
+        # On Windows, we need to find the libs directory regardless of whether or
+        # not its a shared build, and .DLLs are next to the executable, so search
+        # for static first.
+        if not self.spec.satisfies("+shared") or self.spec.satisfies("platform=windows"):
```

Comment:
The change you have here seems pretty close to what we want (good catch!) - the default `.libs` handler (in `spec.py`) specifically searches for `.lib` files regardless of whether the package was built `+shared` or not (because we decided that for windows, `.libs` is specifically for returning the link-time libraries). To that end, I think we actually want to drop `shared_libs` on windows and just search for static libs (I'll think more on this and state a preference tomorrow).

Reply:
To confirm, on Windows

* this should be just `static_libs`, not `static_libs + shared_libs`
* and it might be worth making a comment similar to what is in `_libs_default_handler` in `spec.py` (i.e. `.libs` is about libraries used for linking)
* and IMO it's worth raising an exception if none are found

Reply:
An exception is already raised if none are found.